src/mongo/services/MongoCodeLensProvider.ts

```diff
@@ -13,55 +15,66 @@ export class MongoCodeLensProvider implements vscode.CodeLensProvider {
 
 	public onDidChangeCodeLenses = this._onDidChangeEmitter.event;
 
+	public constructor(private _reporter: TelemetryReporter, private _output: vscode.OutputChannel) {
+		// Empty
+	}
+
 	public setConnectedDatabase(database: string | undefined) {
 		this._connectedDatabase = database;
 		this._connectedDatabaseInitialized = true;
 		this._onDidChangeEmitter.fire();
 	}
 
 	public provideCodeLenses(document: vscode.TextDocument, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.CodeLens[]> {
-		let isInitialized = this._connectedDatabaseInitialized;
-		let isConnected = !!this._connectedDatabase;
-		let database = isConnected && this._connectedDatabase;
-		let lenses: vscode.CodeLens[] = [];
+		const me = this;
```

Comment:
Is `me` and abbreviation for something? I'm not too sure if the `this` variable stands for anything aside from MongoCodeLensProvider.

Reply:
This is a disadvantage of the current implementation of IActionContext.  Since we're getting an action context passed in as "this", we can't use the normal arrow functions and this functionality of TypeScript.  I'm falling back to a standard JavaScript pattern - same the pointer to the class instance to "me" so we can access it instead of "this".

https://stackoverflow.com/questions/13364503/javascript-me-this-why

Reply:
(TypeScript fixes "this" in arrow functions)

Reply:
Thanks for the link. I've faced this problem, but I usually name it according to class. Didn't know it had a name. 

If I understand correctly, your way of doing it is a nicer, standardized way of `var that = this` to be used inside a callback/function scope.

I don't fully agree on the name `me`, but I wouldn't want to force naming a hardly used variable. 
Suggestion: Can you annotate the type of `me` so that it's clearer what to expect from `me`?

Reply:
this vs me is a matter of preference but I've seen me a lot and I think it's more descriptive.

Reply:
Added:

const me: MongoCodeLensProvider = this;